dolphin/api/v1/pools.py

```diff
@@ -12,19 +12,88 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+
+import six
+from webob import exc
+
+from dolphin import db, exception
+from dolphin.api import api_utils
 from dolphin.api.common import wsgi
+from dolphin.api.views import pools as pool_view
+from dolphin.i18n import _
 
 
 class PoolController(wsgi.Controller):
+    def __init__(self):
+        super(PoolController, self).__init__()
+        self.search_options = ['name', 'status', 'id', 'storage_id']
+
+    def _is_storage_valid(self, context, pool, storage_id):
+        """verify the storage and pool association  ."""
+        try:
+            if pool.storage_id != storage_id:
+                return False
+            else:
+                db.storage_get(context, storage_id)
+        except exception.StorageNotFound:
+            msg = _("storage  '%s' not found.") % storage_id
+            raise exc.HTTPNotFound(explanation=msg)
+
+    def _get_pools_search_options(self):
+        """Return pools search options allowed ."""
+        return self.search_options
+
+    def _show(self, req, pool_id, storage_id=None):
+        ctxt = req.environ['dolphin.context']
+        try:
+            pool = db.pool_get(ctxt, pool_id)
+            if storage_id:
```

Comment:
optimization suggestion:
can be avoided multiple if stments:
EX:
if storage_id and if not self._is_storage_valid(ctxt, pool, storage_id):

Reply:
done